src/coreclr/jit/objectalloc.cpp

```diff
@@ -682,6 +818,8 @@ bool ObjectAllocator::CanLclVarEscapeViaParentStack(ArrayStack<GenTree*>* parent
             case GT_EQ:
             case GT_NE:
             case GT_NULLCHECK:
+            case GT_ARR_LENGTH:
+            case GT_INDEX_ADDR:
```

Comment:
```suggestion
```

I do not think we will see a local we are interested in used as an index.

Reply:
Actually, we are of course interested in the array (for some reason I was only thinking about the index) -- but the value of the array local by definition escapes if `GT_INDEX_ADDR` escapes. I think this should be handled in the same way as `GT_ADD` or `GT_FIELD_ADDR` are handled.

Reply:
You should add a test that fails with the current approach (e.g. something that returns the address of an element of a newly allocated array).

Reply:
> I think this should be handled in the same way as GT_ADD or GT_FIELD_ADDR are handled.

Resolved.

> You should add a test that fails with the current approach

I will add test cases later.